### DISCUSSION TITLE
How to specify a one item list with a comma in it?

### DISCUSSION THREAD

**Discussion by wshanks** [tox-dev/tox#3162] [Q&A]
In tox.ini, if I put the following:

```
[tox]
requires =
     tox>=4.6.4,<5
```

I get a parsing error:

```
packaging.requirements.InvalidRequirement: Expected package name at the start of dependency specifier                                                                                                                                         
    <5 
```

It seems that it is treating the `requires` content as a comma delimited list. If instead I use:

```
[tox]
requires =
     tox>=4.6.4,<5
     tox>=4.6.4,<5
```

tox runs normally because now it is treating the entry as a newline delimited list and not interpreting the commas. Is there a way to make a one item list with a comma in it without having the comma interpreted?

**Comment by gaborbernat** [comment]
This is a known behavior of the INI format parser — single-line lists are split on commas. The workaround you found (duplicating the line to force newline-delimited parsing) works, but a cleaner approach is to quote nothing and just ensure newline separation:

```ini
[tox]
requires =
    tox>=4.6.4,<5
```

If this still splits on the comma even with newline format, this would be a bug in the list parser. However, with `tox.toml` this is not an issue since TOML has proper array syntax:

```toml
[tox]
requires = ["tox>=4.6.4,<5"]
```

**Comment by gaborbernat** [comment]
Converted to issue https://github.com/tox-dev/tox/issues/3861
